### ISSUE TITLE
Documentation for sync from upstream trunk

### ISSUE THREAD

**Issue by YasharF** [Entware-ng/Entware-ng#96]
Hi folks,
What is the new process for syncing from openwrt trunk?  I am working on switching the out-of-band security package builds of Entware to use Entware-ng and noticed that my old process to sync from upstream doesn't work anymore.

In the old process, I modified the  OPENWRT_REVISION variable in entware/config.mk (https://github.com/Entware/entware/blob/master/config.mk) which was consumed by buildroot/Makefile to download the trunk svn at the specific version.  In Entware-ng I can't find config.mk or a OPENWRT_REVISION , and realized that Entware-ng seems to already incorporate all of the upstream makefiles in its repo and then there is a LastSync.txt file the is stating when the sync was.

Are you using a script or something that syncs from the upstream, or can you point me to how to do a sync from the upstream so I can restart the out-of-band builds for security related updates in EntwareSecUp?  I already have the feeds updated, just need to figure out the trunk portion of openwrt sync.

(FYI: the new EntwareSecup/Entware-ng https://github.com/EntwareSecUp/Entware-ng/wiki and This is how my old change looked like: https://github.com/EntwareSecUp/entware/commit/905bdee152bd2a585498f14e8aff5c2d836ed446 )

Thanks

**Comment by zyxmon** [comment]
My approach is - generate diff from trunk or a feed: `git diff <last_commit from LastSync.txt> HEAD <subdir>` and apply the changes  manualy to entware or entware_feed.

**Comment by YasharF** [comment]
Thanks.  Would me submitting pull requests like the following help out getting the security related changes from OpenWRT released to Entware-ng faster, or would it just help with reducing your workload?
https://github.com/Entware-ng/Entware-ng/pull/101

**Comment by ryzhovau** [comment]
>  Would me submitting pull requests like #101 help out getting the security related changes from OpenWRT released to Entware-ng faster, or would it just help with reducing your workload?

There is no need to send PRs like this. It will double our work as long as we will (or will not) know what part of upstream code is (or isn't) updated. 

>  I am working on switching the out-of-band security package builds of Entware to use Entware-ng and noticed that my old process to sync from upstream doesn't work anymore.

The main goal is security improve, right? We won't change our upstream sync process in a favor of security. Also, I definitely can't say OpenWrt is security heaven:)

I think some additional feed for `opkg` will be better solution for this purpose. Like:
```
# cat /opt/etc/opkg.conf
src/gz entware-ng http://pkg.entware.net/binaries/mipsel
src/gz entware-ng-sec http://some/other/URL/mipsel

dest root /
lists_dir ext /opt/var/opkg-lists
```
So, users, who will add additional `opkg` feed will receive your security updates.

**Comment by YasharF** [comment]
Got it.  It looks like we can continue with how I had everything setup prior to -ng repo migration.  If anyone asks for out-of-band security updates, feel free to point them to https://github.com/EntwareSecUp/Entware-ng/wiki  . I have been asking for user registration there before giving the repo address out since I am only doing monitoring alerts and updates for packages that there are requests for, instead of the entire feed.

I got rid of the pull request in favor of your existing monthly process.

Closing this issue.
